profiling/src/lib.rs

```diff
@@ -800,4 +789,29 @@ extern "C" fn execute_internal(
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_boolean() {
+        let truthies = [
+            "1", "on", "yes", "true", "On", "Yes", "True", "ON", "YES", "TRUE",
+        ];
+        for val in truthies {
+            assert_eq!(Some(true), parse_boolean(val), "Testing {}", val);
+        }
+
+        let falsies = [
+            "0", "no", "off", "false", "No", "Off", "False", "NO", "OFF", "FALSE",
+        ];
+        for val in falsies {
+            assert_eq!(Some(false), parse_boolean(val), "Testing {}", val);
+        }
+
+        let non_boolean = ["a", "2", "-1", "truuue", "💯", "🦀!", "🔥🔥"];
```

Comment:
```suggestion
        let non_boolean = ["", "a", "2", "-1", "truuue", "💯", "🦀!", "🔥🔥"];
```
Add empty value as it's special cased in the parsing function.